core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -471,6 +473,8 @@ public boolean apply(IndexWarmersMetaData.Entry warmer) {
     /**
      * Returns indexing routing for the given index.
      */
+    // TODO: This can be moved to IndexNameExpressionResolver too, but this means that we will support wildcards and other expressions
+    // in the index,bulk,update and delete apis.
```

Comment:
wouldn't it be possible to move this method where it belongs without supporting wildcards against it though? those apis shouldn't support wildcards